src/poetry/puzzle/provider.py

```diff
@@ -572,7 +572,7 @@ def complete_package(
                 dependency_package = DependencyPackage(
                     dependency,
                     self._pool.package(
-                        package.pretty_name,
+                        package.name,
```

Comment:
Obviously, there is no test, but I expect this to cause some name changes in the lock file (again). (See other comment.) I'd like to avoid that if there is no good reason.

Reply:
why would this cause a change in the lock file?

Reply:
Because the pretty name is "lost" and thus the canonicalized name is written in the lock file.